Yoast/wordpress-seo
# Remove endpoint factory and exceptions that are no longer used

## Context


* In an effort to cleanup the inc directory we're removing classes that are not used at all and have no reasonable use-case to be used by others without deprecating. This class, and the exceptions removed, were only used in the code removed in this PR: https://github.com/Yoast/wordpress-seo/pull/15941

## Summary


This PR can be summarized in the following changelog entry:

* Removes the endpoint factory class as it's not used.
* Removes several exceptions that were not used.

## Relevant technical choices:

* All classes were removed without deprecation as the code behind them has never been truly used.

## Test instructions

This PR can be tested by following these steps:

* There should be no PHP errors or warnings anywhere.
* This code wasn't used anywhere at all so unfortunately there's no specific place to test.
* I'd recommend visiting the frontend, the admin dashboard, the post edit screen, the post overview and the Yoast dashboard to test.

## UI changes

## Documentation

## Quality assurance


Fixes part of https://github.com/Yoast/wordpress-seo/issues/15888
